Auto enable vkb enable portability

gpx1000 opened the pull request:
## Description

Attempt to make the Framework requirement of enabling the Portability extension if on an OSX device and running Vulkan greater than or equal to 1.3 (when the Portability extension is introduced).  Without this, the moltenvk instance wont be detected in instance creation.  Please check instance.cpp for details.

Fixes #476
## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [x] I have tested the sample on at least one compliant Vulkan implementation
- [x] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [x] Any dependent assets have been merged and published in downstream modules
- [x] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)

gpx1000 committed:
Fix issue #452

gpx1000 committed:
Merge remote-tracking branch 'upstream/master' into upstream-master

gpx1000 committed:
Attempt some logic to make setting of VKB_ENABLE_PORTABILITY happen if and only if Vulkan_VERSION is at least 1.3 which should include the required portability extension which will enable the instance to find the non-conformant vulkan for moltenVK.

gpx1000 committed:
update copyright.

TomAtkinsonArm reviewed (dismissed):
LGTM

Tested on Monterey 12.4

gpx1000 committed:
make the compile definition explicit.

gpx1000 committed:
Validated as working with Vulkan 1.3.216.0 on MacBook Air M2.

gpx1000 committed:
enable vulkan.hpp samples to work.

gpx1000 committed:
don't crash open_gl_interop.cpp if a gl_context doesn't get created.  As in the case when extensions required are not present.

gpx1000 committed:
revert the 3.19 cmake version upgrade for platforms other than OSX.
NB: This *might* mean that Android on OSX would have an issue due to the version of CMake that Android SDK might ship with is 3.18.  Revisit this once most people are on Vulkan 1.3 and/or confirmation that android on OSX is a problem.

gpx1000 committed:
revert the 3.19 cmake version upgrade for platforms other than OSX.
NB: This *might* mean that Android on OSX would have an issue due to the version of CMake that Android SDK might ship with is 3.18.  Revisit this once most people are on Vulkan 1.3 and/or confirmation that android on OSX is a problem.

gpx1000 committed:
update the Copyright

gpx1000 committed:
update with clang-format

gpx1000 committed:
oddly running clang-format locally resulted in some white space changes which don't agree with the CICD white space results.

marty-johnson59 commented:
Merging as this is a simple fix for a significant blocking issue.  LMK if there are any problems - we can revert if needed.
